README.md

```diff
@@ -268,6 +276,88 @@ const tx = new SafeTransaction({
   // ...
 })
 const txResponse = await safeSdk.executeTransaction(tx)
+await txResponse.wait()
+```
+
+### buildContractCall
+
+Returns a Safe transaction with the encoded data of a specific contract call ready to be signed and executed.
+
+#### Transaction to enable a module.
+
+```js
+import { SafeSettings } from 'safe-core-sdk'
+
+const tx = await safeSdk.buildContractCall({
+  method: SafeSettings.ENABLE_MODULE,
+  moduleAddress
+})
+```
+
+#### Transaction to disable a Safe module.
+
+```js
+const tx = await safeSdk.buildContractCall({
+  method: SafeSettings.DISABLE_MODULE,
+  moduleAddress
+})
```

Comment:
I'm curious why you chose the approach of having one generic method to perform safe contract calls. It feels too "raw" or low-level 🤔 I think it's more handy to have a method responsible for the call (enableModule, disableModule, etc). This buildContractCall could be used in the internal realization. You can find an example here: https://github.com/gnosis/safe-apps-sdk/blob/master/packages/safe-apps-sdk/src/eth/index.ts#L84 Probably, not the best one 🙈 I also found that in web3.js repo, they also use that approach to build methods for rpc calls

Reply:
This is a good comment :)

I had it implemented that way before refactoring to just just have one generic method. You can check here the commit and the methods I had previously:
https://github.com/gnosis/safe-core-sdk/pull/19/commits/35c0aa17107f10401434ca0230861048b673a7f1#diff-9d1172cb852904ea345f63e4470d48e4b305c14b39b6ef97202c28c356e58676L25

So it is this:

```js
getEnableModuleTx(moduleAddress: string): Promise<SafeTransaction>
getDisableModuleTx(moduleAddress: string): Promise<SafeTransaction>
getAddOwnerTx(ownerAddress: string, threshold?: number): Promise<SafeTransaction>
getRemoveOwnerTx(ownerAddress: string, threshold?: number): Promise<SafeTransaction>
getSwapOwnerTx(oldOwnerAddress: string, newOwnerAddress: string): Promise<SafeTransaction>
getChangeThresholdTx(threshold: number): Promise<SafeTransaction>
```

vs. this:

```js
buildContractCall(params: ContractCallParams): Promise<SafeTransaction>
```

I don't really have a strong opinion on this. I unified them because I thought the logic/concept (not the implementation) was almost the same: to get the encoding of a contract call data. So why not using just one function passing the contract method name as an argument?

Does this look too low-level? It may be.
What do you think @rmeissner? Do you also agree that is easier for the devs to have one independent method for each contract call?

Reply:
Hmm I also don't have a strong opinion. Ethers allows you to do both approaches, right? Not sure what is best. Probably the one where autocomplete works the best :P